File: Java/src/1342_NumberOfStepsToReduceANumberToZero/NumberOfStepsToReduceANumberToZero.java
class Solution {
    public int numberOfSteps (int num) {
        int ans = 0 ;
        while(num > 0){
            if(num % 2 == 0){
                ans++;
                num/= 2 ;
            }else {
                ans++;
                num--;
            }
        }
        return ans;
    }
}

File: golang/1342_NumberOfStepsToReduceANumberToZero/numberOfStepsToReduceANumberToZero.go
package numberofstepstoreduceanumbertozero

func numberOfSteps(num int) int {
	ans := 0
	for num != 0 {
		if num%2 == 0 {
			num /= 2
		} else {
			num--
		}
		ans++
	}
	return ans
}
